subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/LazyDownloadsIntegrationTest.groovy

```diff
@@ -67,13 +66,14 @@ class LazyDownloadsIntegrationTest extends AbstractHttpDependencyResolutionTest
         succeeds("graph")
     }
 
-    @ToBeFixedForConfigurationCache(because = "Uses Configuration API in task action")
-    def "downloads only the metadata when resolved artifacts are queried"() {
+    def "downloads only the metadata when all components are queried"() {
         given:
         buildFile << """
+            def allComponents = configurations.compile.incoming.resolutionResult.allComponents
```

Comment:
This is doing resolution at configuration time.  I wouldn't trade CC compatibility for this for now.